File: Scripts/results/generated_code_results/baseline_gpt/Education/OnlineCourseMobileApp/DiscoverScreen.js
import React from 'react';
import { View, Text, TextInput, TouchableOpacity, FlatList, StyleSheet, Image } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const DiscoverScreen = () => {
  // Sample data
  const courses = [
    { id: '1', title: 'How to make modern poster for Beginner', category: 'GRAPHICS DESIGN', price: '$12.00' },
    { id: '2', title: 'How to make Design system in easy waen', category: 'UI/UX DESIGN', price: '$12.00' },
  ];

  const renderCourseItem = ({ item }) => (
    <View style={styles.courseCard}>
      <Image source={require('../assets/snack-icon.png')} style={styles.courseImage} />
      <Text style={styles.courseCategory}>{item.category}</Text>
      <Text style={styles.courseTitle}>{item.title}</Text>
      <Text style={styles.coursePrice}>{item.price}</Text>
      <View style={styles.ratingContainer}>
        <Icon name="star" size={16} color="#F5A623" />
        <Text style={styles.ratingText}>4.9</Text>
      </View>
    </View>
  );

  return (
    <View style={styles.container}>
      <Text style={styles.heading}>Discover the best {"\n"}course that suits to you</Text>
      <View style={styles.searchContainer}>
        <TextInput style={styles.searchInput} placeholder="Search course..." />
        <TouchableOpacity style={styles.filterButton}>
          <Icon name="filter-outline" size={24} color="#000" />
        </TouchableOpacity>
      </View>
      
      <View style={styles.sectionHeader}>
        <Text style={styles.sectionTitle}>Course of the weeks</Text>
        <TouchableOpacity>
          <Text style={styles.seeAllText}>See all</Text>
        </TouchableOpacity>
      </View>
      
      <FlatList
        data={courses}
        renderItem={renderCourseItem}
        keyExtractor={item => item.id}
        horizontal
        showsHorizontalScrollIndicator={false}
        style={styles.courseList}
      />

      <View style={styles.sectionHeader}>
        <Text style={styles.sectionTitle}>Suits to you</Text>
        <TouchableOpacity>
          <Text style={styles.seeAllText}>See all</Text>
        </TouchableOpacity>
      </View>

      <FlatList
        data={courses}
        renderItem={renderCourseItem}
        keyExtractor={item => item.id}
        horizontal
        showsHorizontalScrollIndicator={false}
        style={styles.courseList}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    padding: 16,
  },
  heading: {
    fontSize: 24,
    fontWeight: 'bold',
    marginVertical: 16,
  },
  searchContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 20,
  },
  searchInput: {
    flex: 1,
    height: 40,
    borderRadius: 20,
    backgroundColor: '#f4f4f4',
    paddingHorizontal: 20,
  },
  filterButton: {
    marginLeft: 10,
    backgroundColor: '#f4f4f4',
    borderRadius: 10,
    padding: 8,
  },
  sectionHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginVertical: 10,
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  seeAllText: {
    fontSize: 14,
    color: '#007BFF',
  },
  courseList: {
    paddingVertical: 10,
  },
  courseCard: {
    width: 160,
    padding: 10,
    marginRight: 16,
    backgroundColor: '#fff',
    borderRadius: 10,
    shadowColor: '#000',
    shadowOpacity: 0.1,
    shadowRadius: 10,
    elevation: 2,
  },
  courseImage: {
    width: '100%',
    height: 80,
    borderRadius: 5,
  },
  courseCategory: {
    marginTop: 10,
    fontSize: 12,
    color: '#aaa',
  },
  courseTitle: {
    marginTop: 5,
    fontSize: 14,
    fontWeight: 'bold',
  },
  coursePrice: {
    marginTop: 5,
    fontSize: 16,
    color: '#0AADA8',
  },
  ratingContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 5,
  },
  ratingText: {
    marginLeft: 5,
    fontSize: 14,
    color: '#333',
  },
});

export default DiscoverScreen;

File: Scripts/results/generated_code_results/Flutter/Education/OnlineCourseMobileApp/DiscoverScreen.dart
import 'package:flutter/material.dart';
class DiscoverScreen extends StatelessWidget {
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Text('Discover the best course that suits to you'),
actions: [
Stack(
children: [
IconButton(
icon: Icon(Icons.notifications),
onPressed: () {},
),
Positioned(
right: 8,
top: 8,
child: Container(
padding: EdgeInsets.all(2),
decoration: BoxDecoration(
color: Colors.red,
borderRadius: BorderRadius.circular(10),
),
constraints: BoxConstraints(
minWidth: 16,
minHeight: 16,
),
child: Text(
'1',
style: TextStyle(
color: Colors.white,
fontSize: 10,
),
textAlign: TextAlign.center,
),
),
),
],
),
],
),
body: SingleChildScrollView(
child: Padding(
padding: EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
TextField(
decoration: InputDecoration(
hintText: 'Search course...',
prefixIcon: Icon(Icons.search),
border: OutlineInputBorder(
borderRadius: BorderRadius.circular(30),
),
),
onSubmitted: (value) {
Navigator.pushNamed(context, 'Search');
},
),
SizedBox(height: 24),
Row(
mainAxisAlignment: MainAxisAlignment.spaceBetween,
children: [
Text(
'Course of the weeks',
style: Theme.of(context).textTheme.titleLarge,
),
TextButton(
child: Text('See all'),
onPressed: () {},
),
],
),
SizedBox(height: 16),
SingleChildScrollView(
scrollDirection: Axis.horizontal,
child: Row(
children: [
_buildCourseCard(
context,
'How to make modern poster for Beginner',
'GRAPHICS DESIGN',
4.9,
'\$12.00',
),
SizedBox(width: 16),
_buildCourseCard(
context,
'How to make Design system in easy waen',
'UI/UX DESIGN',
4.9,
'\$12.00',
),
],
),
),
SizedBox(height: 24),
Row(
mainAxisAlignment: MainAxisAlignment.spaceBetween,
children: [
Text(
'Suits to you',
style: Theme.of(context).textTheme.titleLarge,
),
TextButton(
child: Text('See all'),
onPressed: () {},
),
],
),
SizedBox(height: 16),
SingleChildScrollView(
scrollDirection: Axis.horizontal,
child: Row(
children: [
_buildCourseCard(
context,
'How to make modern poster',
'GRAPHICS DESIGN',
4.9,
null,
),
SizedBox(width: 16),
_buildCourseCard(
context,
'How to make Design system',
'UI/UX DESIGN',
4.9,
null,
),
],
),
),
],
),
),
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: 1,
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),
BottomNavigationBarItem(icon: Icon(Icons.explore), label: 'Discover'),
BottomNavigationBarItem(icon: Icon(Icons.book), label: 'My Courses'),
BottomNavigationBarItem(icon: Icon(Icons.person), label: 'Profile'),
],
onTap: (index) {
if (index == 0) {
Navigator.pushNamed(context, 'Home');
} else if (index == 2) {
Navigator.pushNamed(context, 'MyCourses');
}
},
),
);
}
Widget _buildCourseCard(BuildContext context, String title, String type, double rating, String? price) {
return GestureDetector(
onTap: () {
Navigator.pushNamed(context, 'CourseDetail');
},
child: Container(
width: 200,
child: Card(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Image.asset('assets/images/template.png', height: 100, width: 200, fit: BoxFit.cover),
Padding(
padding: EdgeInsets.all(8.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(type, style: TextStyle(fontSize: 12, color: Colors.grey)),
SizedBox(height: 4),
Text(title, style: TextStyle(fontWeight: FontWeight.bold)),
SizedBox(height: 4),
Row(
children: [
Icon(Icons.star, color: Colors.yellow, size: 16),
SizedBox(width: 4),
Text('$rating', style: TextStyle(fontSize: 12)),
],
),
if (price != null) SizedBox(height: 4),
if (price != null) Text(price, style: TextStyle(fontWeight: FontWeight.bold)),
],
),
),
],
),
),
),
);
}
}